<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- SETUP ARGUMENTS -->
    <!-- enable/disable rviz -->
    <arg name="rviz" default="false" />
    <!-- enable/disable calibration -->
    <arg name="calibrate" default="false"/>

    <arg name="camera_name" default="deimos"/>
 
    <!-- SETUP ROBOT MODEL -->
    <!-- (robot model should be handelled by master) -->
    <param name="$(arg camera_name)_robot_description" command="$(find xacro)/xacro '$(find i3dr_deimos)/urdf/deimos_scene.urdf.xacro'" />
    <arg name="use_tf_static" default="false"/>
    <node name="robot_state_publisher_deimos" pkg="robot_state_publisher" type="state_publisher">
        <remap from="robot_description" to="$(arg camera_name)_robot_description"/>
        <param name="use_tf_static" value="$(arg use_tf_static)"/>
    </node>

    <!-- STEREO CAPTURE -->
    <include file="$(find i3dr_deimos)/launch/deimos_capture.launch">
        <arg name="camera_namespace" value="$(arg camera_name)"/>
        <arg name="device" value="/dev/video0"/>
        <arg name="exposure" default="5000" />
    </include>

    <!-- STEREO CALIBRATION -->
    <include if="$(arg calibrate)" file="$(find i3dr_stereo_camera)/launch/stereo_calibration.launch">
        <arg name="camera_namespace" value="$(arg camera_name)"/>
    </include>

    <!-- STEREO MATCHER -->
    <include file="$(find i3dr_stereo_camera)/launch/stereo_matcher.launch">
        <arg name="use_i3dr_matcher" value="true"/>

        <arg name="camera_namespace" value="$(arg camera_name)"/>
        <arg name="camera_name" value="$(arg camera_name)"/>
        <arg name="gui" value="false"/>
        <arg name="stereo_algorithm" value="1" />
        <arg name="min_disparity" value="-18" />
        <arg name="disparity_range" value="96" />
        <arg name="correlation_window_size" value="19" />
        <arg name="uniqueness_ratio" value="2" />
        <arg name="speckle_size" value="1000" />
        <arg name="prefilter_size" value="147" />
        <arg name="prefilter_cap" value="7" />
    </include>

    <!-- DISPLAY -->
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rivz" 
        args="-d $(find i3dr_deimos)/rviz/scene.rviz"/>

</launch>